lib/journey-model.js

```diff
@@ -0,0 +1,35 @@
+'use strict';
+
+const SessionModel = require('./model');
+const _ = require('underscore');
+
+class JourneyModel extends SessionModel {
+
+    registerModel(model) {
+        let modelNames = this.get('registered-models') || [];
+        if (!_.contains(modelNames, model.options.key)) {
+            modelNames.push(model.options.key);
+            this.set('registered-models', modelNames);
+        }
+        this.currentModels = this.currentdModels || [];
```

Comment:
registering needs to be tested twice, it defaults to empty so it works first time.